play/src/main/scala/caliban/PlayJson.scala

```diff
@@ -0,0 +1,123 @@
+package caliban
+
+import caliban.Value._
+import caliban.parsing.adt.LocationInfo
+import play.api.http.Writeable
+import play.api.libs.json.Reads._
+import play.api.libs.json.Writes._
+import play.api.libs.json._
+
+import scala.util.Try
+
+object PlayJson {
+
+  //// JSON Reads
+  private def jsonToInputValue(json: JsValue): InputValue = json match {
+    case JsNull             => NullValue
+    case boolean: JsBoolean => BooleanValue(boolean.value)
+    case JsNumber(value) =>
+      Try(value.toIntExact).fold(_ => FloatValue(value), IntValue(_))
+    case JsString(value) => StringValue(value)
+    case JsArray(array) =>
+      InputValue.ListValue(array.toList.map(jsonToInputValue))
+    case JsObject(obj) =>
+      val builder = Predef.Map.newBuilder[String, InputValue]
+      builder.sizeHint(obj)
+      obj.foreach { case (k, v) => builder += (k -> jsonToInputValue(v)) }
```

Comment:
I generally prefer `foldLeft` for this. i.e.

```scala
InputValue.ObjectValue(obj.foldLeft(builder) { case (k, v) => builder += (k -> jsonToInputValue(v)) }.result())

```

Reply:
My intention was to make it very loud and clear that there's a side effect happening, however local and contained it is, but it's not a big deal to change too.